### PR TITLE
jless: init at 0.7.1

### PR DESCRIPTION
**Pull request by jfchevrette**
###### Motivation for this change

Adds JSON pager `jless` to nixpkgs

https://github.com/PaulJuliusMartinez/jless

Also adds myself to all-maintainers and as a maintainer of this package

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/jless/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "PaulJuliusMartinez";
+    repo = "jless";
+    rev = "v${version}";
+    sha256 = "sha256-gBqyo/N/qF6HCTUrSKNVLiL1fc/JTfip1kNpNCBzRT8=";
+  };
+
+  cargoSha256 = "sha256-PbX61RVbrI2kTuyXK+LhQdJDvNo3KjIQH5eBbL6iUBM=";
+
+  doCheck = false;
```
Comment (zowoq) [inline pkgs/development/tools/jless/default.nix:16]: If it is only a couple of tests that fail they can be selectively disabled via `checkFlags`.

https://github.com/NixOS/nixpkgs/blob/00be9c5821622fefc6da17a664b1a536a5cbe5f7/pkgs/tools/misc/miniserve/default.nix#L28-L31

If a lot of the tests fail `doCheck = false` is fine but please add a comment.

File: pkgs/development/tools/jless/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with lib; {
+    description = "A command-line pager for JSON data.";
```
Comment (zowoq) [inline pkgs/development/tools/jless/default.nix:19]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

`meta.description` shouldn't end with a period.

```suggestion
    description = "A command-line pager for JSON data";
```

File: pkgs/development/tools/jless/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    homepage = "https://github.com/PaulJuliusMartinez/jless";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jfchevrette ];
+    mainProgram = "jless";
```
Comment (zowoq) [inline pkgs/development/tools/jless/default.nix:23]: Redundant as it matches `pname`.

```suggestion
```

File: pkgs/development/tools/jless/default.nix (hunk 1 above)
Comment (jfchevrette) [reply pkgs/development/tools/jless/default.nix]: Good catch. This was actually based off of another similar derivation and I forgot to remove `doCheck = false`. The tests actually all pass so I removed the line. Thanks!

File: pkgs/development/tools/jless/default.nix (hunk 2 above)
Comment (jfchevrette) [reply pkgs/development/tools/jless/default.nix]: Fixed!

File: pkgs/development/tools/jless/default.nix (hunk 3 above)
Comment (jfchevrette) [reply pkgs/development/tools/jless/default.nix]: Removed. I read a bit more on meta.mainProgram and I understand it better now. Thanks for pointing it out!

**Pull request merged**